src/shims/unix/fs.rs

```diff
@@ -559,8 +559,14 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
                 );
             };
 
-            if mode != 0o666 {
-                throw_unsup_format!("non-default mode 0o{:o} is not supported", mode);
+            match this.tcx.sess.target.os.as_ref() {
+                // Support all modes on UNIX host
```

Comment:
Comment and code do not match. The comment tests whether the *target* is unix or Windows. It will always be Unix here.

You need to check the *host*, via `#[cfg]`/`cfg!`.